Add changelog for homeless-shelter-to-proc (#11350)
NixOS/nix: doc/manual/rl-next/homeless-shelter-to-proc.md

---
synopsis: On linux, set $HOME=/proc/homeless-shelter instead of /homeless-shelter
issues: [8313, 11295]
prs: [11300]
---

> When building, $HOME is set to a non-existing dir. Previously it was always set to `/homeless-shelter`. After a build, the builder verifies that it doesn't exist. In some scenarios (specifically when using the linux sandbox with a single-user installation), it is possible to create the `/homeless-shelter` directory, and some tools will create it, resulting in a build error.

Reviewer: ```suggestion
When building, $HOME is set to a non-existing directory. Previously it was always set to `/homeless-shelter`. Before a build, Nix verifies that it doesn't exist. In some scenarios (specifically when using the Linux sandbox with a single-user installation), it is possible to create the `/homeless-shelter` directory, and some tools will create it, resulting in a build error.
```
